cmd/thanos/sidecar.go

```diff
@@ -262,43 +263,52 @@ func runSidecar(
 
 		exemplarSrv := exemplars.NewPrometheus(conf.prometheus.url, c, m.Labels)
 
-		infoSrv := info.NewInfoServer(
-			component.Sidecar.String(),
-			info.WithLabelSetFunc(func() []labelpb.ZLabelSet {
+		var infoOptions []info.ServerOptionFunc
+		if !promAgentModeEnabled {
+			infoOptions = append(infoOptions, info.WithLabelSetFunc(func() []labelpb.ZLabelSet {
 				return promStore.LabelSet()
 			}),
-			info.WithStoreInfoFunc(func() *infopb.StoreInfo {
-				if httpProbe.IsReady() {
-					mint, maxt := promStore.Timestamps()
-					return &infopb.StoreInfo{
-						MinTime:                      mint,
-						MaxTime:                      maxt,
-						SupportsSharding:             true,
-						SupportsWithoutReplicaLabels: true,
-						TsdbInfos:                    promStore.TSDBInfos(),
+				info.WithStoreInfoFunc(func() *infopb.StoreInfo {
+					if httpProbe.IsReady() {
+						mint, maxt := promStore.Timestamps()
+						return &infopb.StoreInfo{
+							MinTime:                      mint,
+							MaxTime:                      maxt,
+							SupportsSharding:             true,
+							SupportsWithoutReplicaLabels: true,
+							TsdbInfos:                    promStore.TSDBInfos(),
+						}
 					}
-				}
-				return nil
-			}),
-			info.WithExemplarsInfoFunc(),
+					return nil
+				}),
+				info.WithExemplarsInfoFunc())
+		}
+		infoOptions = append(infoOptions,
 			info.WithRulesInfoFunc(),
 			info.WithTargetsInfoFunc(),
 			info.WithMetricMetadataInfoFunc(),
 		)
+		infoSrv := info.NewInfoServer(
+			component.Sidecar.String(), infoOptions...,
+		)
 
-		storeServer := store.NewLimitedStoreServer(store.NewInstrumentedStoreServer(reg, promStore), reg, conf.storeRateLimits)
-		s := grpcserver.New(logger, reg, tracer, grpcLogOpts, tagOpts, comp, grpcProbe,
-			grpcserver.WithServer(store.RegisterStoreServer(storeServer, logger)),
-			grpcserver.WithServer(rules.RegisterRulesServer(rules.NewPrometheus(conf.prometheus.url, c, m.Labels))),
+		opts := []grpcserver.Option{
 			grpcserver.WithServer(targets.RegisterTargetsServer(targets.NewPrometheus(conf.prometheus.url, c, m.Labels))),
 			grpcserver.WithServer(meta.RegisterMetadataServer(meta.NewPrometheus(conf.prometheus.url, c))),
-			grpcserver.WithServer(exemplars.RegisterExemplarsServer(exemplarSrv)),
 			grpcserver.WithServer(info.RegisterInfoServer(infoSrv)),
 			grpcserver.WithListen(conf.grpc.bindAddress),
 			grpcserver.WithGracePeriod(conf.grpc.gracePeriod),
 			grpcserver.WithMaxConnAge(conf.grpc.maxConnectionAge),
 			grpcserver.WithTLSConfig(tlsCfg),
-		)
+		}
+		if !promAgentModeEnabled {
+			storeServer := store.NewLimitedStoreServer(store.NewInstrumentedStoreServer(reg, promStore), reg, conf.storeRateLimits)
+			opts = append(opts, grpcserver.WithServer(store.RegisterStoreServer(storeServer, logger)),
```

Comment:
Just a bit easier to read
```suggestion
			opts = append(opts, 
				grpcserver.WithServer(store.RegisterStoreServer(storeServer, logger)),
```